src/transaction.rs

```diff
@@ -137,3 +151,139 @@ pub fn remove_spent_notes(
     }
     serde_wasm_bindgen::to_value(&unspent_notes).expect("Cannot serialize unspent notes")
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct JSTransaction {
+    pub txid: String,
+    pub txhex: String,
+    pub nullifiers: Vec<String>,
+}
+
+#[wasm_bindgen]
+pub fn create_transaction(
+    notes: JsValue,
+    extsk: &str,
+    to_address: &str,
+    change_address: &str,
+    amount: u64,
+    block_height: u32,
+    is_testnet: bool,
+) -> JsValue {
+    let notes: Vec<(Note, String)> =
+        serde_wasm_bindgen::from_value(notes).expect("Cannot deserialize notes"); // Note, Witness, Address
+    let extsk = decode_extsk(extsk, is_testnet);
+    let network = if is_testnet {
+        Network::TestNetwork
+    } else {
+        Network::MainNetwork
+    };
+    let result = create_transaction_internal(
+        &notes,
+        &extsk,
+        to_address,
+        change_address,
+        amount,
+        BlockHeight::from_u32(block_height),
+        network,
+    )
+    .expect("Failed to create tx");
+    serde_wasm_bindgen::to_value(&result).expect("Cannot serialize transaction")
+}
+
+pub fn create_transaction_internal(
+    notes: &[(Note, String)],
+    extsk: &ExtendedSpendingKey,
+    to_address: &str,
+    change_address: &str,
+    amount: u64,
+    block_height: BlockHeight,
+    network: Network,
+) -> Result<JSTransaction, Box<dyn Error>> {
+    let mut builder = Builder::new(network, block_height);
+
+    let fee = 2365000u64;
+
+    let mut total = 0;
+    let mut nullifiers = vec![];
+    for (note, witness) in notes {
```

Comment:
can we sort the notes by size before looping?

Reply:
We should actually do it in MPW as well